docs/samples/Microsoft.ML.Samples/Dynamic/GAM.cs

```diff
@@ -0,0 +1,82 @@
+                        new TextLoader.Column("NitricOxides", DataKind.R4, 5),
+                        new TextLoader.Column("RoomsPerDwelling", DataKind.R4, 6),
+                        new TextLoader.Column("PercentPre40s", DataKind.R4, 7),
+                        new TextLoader.Column("EmploymentDistance", DataKind.R4, 8),
+                        new TextLoader.Column("HighwayDistance", DataKind.R4, 9),
+                        new TextLoader.Column("TaxRate", DataKind.R4, 10),
+                        new TextLoader.Column("TeacherRatio", DataKind.R4, 11),
+                    }
+                });
+            
+            // Read the data
+            var data = reader.Read(dataFile);
+
+            // Step 2: Pipeline
+            // Concatenate the features to create a Feature vector.
+            // Then append a gam regressor, setting the "MedianHomeValue" column as the label of the dataset, and 
+            // the "Features" column produced by concatenation as the features column.
+            var pipeline = mlContext.Transforms.Concatenate("Features", "CrimesPerCapita", "PercentResidental", "PercentNonRetail", 
+                "CharlesRiver", "NitricOxides", "RoomsPerDwelling", "PercentPre40s", "EmploymentDistance", "HighwayDistance", "TaxRate", "TeacherRatio")
+                    .Append(mlContext.Regression.Trainers.GeneralizedAdditiveModels(labelColumn: "MedianHomeValue", featureColumn: "Features", maxBins: 16));
+            var fitPipeline = pipeline.Fit(data);
+
+            // Extract the model from the pipeline
+            var gamModel = fitPipeline.LastTransformer.Model;
+
+            // Step 3: Investigate the properties of the model
+
+            // The intercept for the GAM models represent the average prediction for the training data
+            var intercept = gamModel.GetIntercept();
+            Console.WriteLine($"Average predicted cost: {intercept}");
+
+            // Each feature represents the deviation from the average prediction as a function of the feature value
+            
+            // Let's take a look at the TeacherRatio
+            var teacherRatioBinUpperBounds = gamModel.GetFeatureBinUpperBounds(10);
+            var teacherRatioFeatureWeights = gamModel.GetFeatureWeights(10);
+
+            Console.WriteLine("We can see that smaller class sizes are predictive of a higher house value. Student-teacher ratios larger than ~18, lead to lower-than-average house price predictions.");
+            Console.WriteLine("Student-Teacher Ratio");
+            for (int i = 0; i < teacherRatioBinUpperBounds.Length; i++)
+            {
+                Console.WriteLine($"x < {teacherRatioBinUpperBounds[i]:0.00} => {teacherRatioFeatureWeights[i]:0.000}");
+            }
+
+            Console.WriteLine("Note that these measurements are noisy (see student-teacher ratios > xyz). Common practice is to use resampling methods to estimate confidence at each bin.");
```

Comment:
Will update the `xyz` placeholder with the correct value once the awaited PR is in.

Reply:
Resolved.